src/database/import.ts

```diff
@@ -0,0 +1,172 @@
+import * as Chain from "stream-chain";
+import * as clc from "colorette";
+import * as Filter from "stream-json/filters/Filter";
+import * as stream from "stream";
+import * as StreamObject from "stream-json/streamers/StreamObject";
+
+import pLimit from "p-limit";
+
+import { URL } from "url";
+import { Client, ClientResponse } from "../apiv2";
+import { FirebaseError } from "../error";
+
+const MAX_CHUNK_SIZE = 1024 * 1024;
+const CONCURRENCY_LIMIT = 5;
+
+type Data = {
+  json: { [key: string]: any } | string | number | boolean;
+  pathname: string;
+};
+
+type ChunkedData = {
+  chunks: Data[] | null;
+  size: number;
+};
+
+/**
+ * Imports JSON data to a given RTDB instance.
+ *
+ * The data is parsed and chunked into subtrees of ~1 MB, to be subsequently written in parallel.
+ */
+export default class DatabaseImporter {
+  private client: Client;
+  private limit = pLimit(CONCURRENCY_LIMIT);
+
+  constructor(
+    private dbUrl: URL,
+    private inStream: stream.Readable,
+    private dataPath: string,
+    private chunkSize = MAX_CHUNK_SIZE
+  ) {
+    this.client = new Client({ urlPrefix: dbUrl.origin, auth: true });
+  }
+
+  /**
+   * Writes the chunked data to RTDB. Any existing data at the specified location will be overwritten.
+   */
+  async execute(): Promise<any> {
```

Comment:
Could you take an another pass through this file and try to clean up all `any`s that are not absolutely necessary? It seems like we should be able to add a real return type for writeChunk since we own the API it is calling & should know what to expect back. Once we add that, we should be able to propogate the real type up through the call stack too.

If there is somewhere in here where we can't know the type, we should use `unknown` instead of `any`.

Reply:
Done.